server/src/main/java/org/elasticsearch/indices/cluster/IndicesClusterStateService.java

```diff
@@ -988,6 +996,29 @@ private static DiscoveryNode findSourceNodeForPeerRecovery(RoutingTable routingT
         return sourceNode;
     }
 
+    private static DiscoveryNode findSourceNodeForReshardSplitRecovery(
+        RoutingTable routingTable,
+        DiscoveryNodes nodes,
+        ShardId sourceShardId
+    ) {
+        ShardRouting sourceShardRouting = routingTable.shardRoutingTable(sourceShardId).primaryShard();
+
+        if (sourceShardRouting.active() == false) {
+            logger.trace("can't find reshard split source node because source shard {} is not active.", sourceShardRouting);
```

Comment:
We should assert false here and below as well.

Reply:
If it fails we can leave it out for now, but then we should add a jira to ensure we have these properties in the routing table.